### PR TITLE
TW-415 Can invite people with a matrix account to direct chat

### PR DESCRIPTION
**Pull request by drminh2807**
# Issue
- #415 

# Demo

https://github.com/linagora/twake-on-matrix/assets/12546908/58385c20-3070-4381-99c1-dd2ad728b209

### CODE REVIEW

File: lib/pages/new_private_chat/search_contacts_controller.dart [hunk 1]
```diff
@@ -1,18 +1,14 @@
...
+import 'package:fluffychat/presentation/converters/presentation_contact_converter.dart';
 import 'package:flutter/material.dart';
 import 'package:matrix/matrix.dart';
 
 mixin class SearchContactsController {
   static const _debouncerIntervalInMilliseconds = 300;
 
-  final _getContactController =
-      GetContactsController(_PresentationContactConverter());
+  late GetContactsController _getContactController;
```
Comment (hoangdat) [inline lib/pages/new_private_chat/search_contacts_controller.dart:11]: prefer nullable

**Review by hoangdat — APPROVED** [review]

**Comment by sherlockvn** [comment]
> # Issue
> * [can invite people with a matrix account to direct chat #415](https://github.com/linagora/twake-on-matrix/issues/415)
> 
> # Demo
>  Screen.Recording.2023-08-15.at.11.32.42.mov

When you click send, why don't the text in textfield is clear ?

File: lib/pages/new_private_chat/widget/external_contact_widget.dart [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                        contact.matrixId ?? "",
+                        style: Theme.of(context).textTheme.titleMedium?.merge(
+                              TextStyle(
+                                overflow: TextOverflow.ellipsis,
+                                letterSpacing: 0.15,
+                                color: Theme.of(context)
+                                    .colorScheme
+                                    .onSurfaceVariant,
+                              ),
+                            ),
```
Comment (nqhhdev) [inline lib/pages/new_private_chat/widget/external_contact_widget.dart:58]: Why do you use `merge`?
Currently, It was full of basic information
Screenshot 2023-08-16 at 09 00 18
IMO: Should use `Theme.of(context).textTheme.titleMedium?.copyWith(color: anything)` and `overflow: TextOverflow.ellipsis` outside `style`

File: lib/pages/new_private_chat/widget/external_contact_widget.dart [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                        contact.email ?? "",
+                        style: Theme.of(context).textTheme.bodyMedium?.merge(
+                              TextStyle(
+                                overflow: TextOverflow.ellipsis,
+                                letterSpacing: 0.15,
+                                color:
+                                    LinagoraRefColors.material().tertiary[30],
+                              ),
+                            ),
+                      ),
```
Comment (nqhhdev) [inline lib/pages/new_private_chat/widget/external_contact_widget.dart:70]: same

File: lib/pages/new_private_chat/new_private_chat_view.dart [hunk 4]
```diff
@@ -26,9 +26,17 @@ class NewPrivateChatView extends StatelessWidget {
...
+        child: controller.contactsNotifier == null
+            ? null
+            : ExpansionList(
+                contactsNotifier: controller.contactsNotifier!,
+                goToNewGroupChat: controller.goToNewGroupChat,
+                isSearchModeNotifier: controller.isSearchModeNotifier,
+                isShowContactsNotifier: controller.isShowContactsNotifier,
+                onContactTap: controller.onContactAction,
+                onExternalContactTap: controller.onExternalContactAction,
+                toggleContactsList: () {},
```
Comment (nqhhdev) [inline lib/pages/new_private_chat/new_private_chat_view.dart:38]: WARNINGGGGGGGGGG, misssssssssssss

`newPrivateChatController.toggleContactsList()`

File: lib/pages/contacts_tab/contacts_tab_body_view.dart [hunk 5]
```diff
@@ -20,72 +20,75 @@ class ContactsTabBodyView extends StatelessWidget {
...
-        child: ValueListenableBuilder(
-          valueListenable: controller.contactsNotifier,
-          builder: (context, value, child) => value.fold(
-            (failure) => const EmptyContactBody(),
-            (success) {
-              if (success is! PresentationContactsSuccess) {
-                return const LoadingContactWidget();
-              }
+        child: controller.contactsNotifier == null
+            ? null
```
[the remark this answers is not in the document]
Comment (nqhhdev) [reply lib/pages/contacts_tab/contacts_tab_body_view.dart:24]: I think it's should apply early returning

File: lib/pages/new_private_chat/new_private_chat_view.dart (hunk 4 above)
Comment (drminh2807) [reply lib/pages/new_private_chat/new_private_chat_view.dart]: done

File: lib/pages/new_private_chat/widget/external_contact_widget.dart (hunk 3 above)
Comment (drminh2807) [reply lib/pages/new_private_chat/widget/external_contact_widget.dart]: done

File: lib/pages/new_private_chat/widget/external_contact_widget.dart (hunk 2 above)
Comment (drminh2807) [reply lib/pages/new_private_chat/widget/external_contact_widget.dart]: done

File: lib/pages/new_group/new_group_view.dart [hunk 6]
```diff
@@ -46,9 +46,15 @@ class NewGroupView extends StatelessWidget {
                     child: child,
                   );
                 },
-                child: ContactsSelectionList(
-                  newGroupController: controller,
-                ),
+                child: controller.contactsNotifier == null
+                    ? null
```
[the remark this answers is not in the document]
Comment (drminh2807) [reply lib/pages/new_group/new_group_view.dart:50]: no need to change

File: lib/pages/contacts_tab/contacts_tab_body_view.dart (hunk 5 above)
[the remark this answers is not in the document]
Comment (drminh2807) [reply lib/pages/contacts_tab/contacts_tab_body_view.dart:24]: no need to change

File: lib/pages/contacts_tab/contacts_tab_body_view.dart [hunk 7]
```diff
@@ -20,72 +20,75 @@ class ContactsTabBodyView extends StatelessWidget {
...
-                    padding: const EdgeInsets.only(left: 8.0, top: 8.0),
-                    child: NoContactsFound(
-                      keyword: success.keyword,
-                    ),
-                  );
-                }
-              }
+                    if (success.data.isEmpty) {
+                      if (success.keyword.isEmpty) {
+                        return const EmptyContactBody();
```
[the remark this answers is not in the document]
Comment (drminh2807) [reply lib/pages/contacts_tab/contacts_tab_body_view.dart:36]: removed above component

**Review by sherlockvn — APPROVED** [review]

**Review by nqhhdev — APPROVED** [review]

**Pull request merged**